sheduling/src/App.js

```diff
@@ -1,20 +1,40 @@
-import React from "react";
+import React, { Component } from "react";
 import "./App.css";
 import Login from "./views/Login";
-import Tasks from "./views/Tasks";
-import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
+import Dashboard from "./views/Dashboard";
+import { BrowserRouter as Router, Route } from "react-router-dom";
+import { connect } from "react-redux";
 
-function App() {
-  return (
-    <Router>
-      <div className="App">
-        <Switch>
-          <Route path="/" exact component={Login} />
-          <Route path="/tasks" component={Tasks} />
-        </Switch>
-      </div>
-    </Router>
-  );
+import AdminWrapper from "./views/AdminWrapper";
+
+class App extends Component {
+  render() {
+    return (
+      <Router>
+        <div className="App">
+          <Route
+            path="/"
+            render={props => {
+              return (
+                <AdminWrapper>
+                  {this.props.auth.token ? <Dashboard /> : <Login />}
+                </AdminWrapper>
+              );
+            }}
+          />
+        </div>
+      </Router>
+    );
+  }
 }
 
-export default App;
+const mapStateToProps = state => {
+  return {
+    auth: state.auth
```

Comment:
This is where you are setting the `auth` prop to match that which is in the store